v21.2/alter-role.md

```diff
@@ -45,18 +45,19 @@ Parameter | Description
-`PASSWORD password`/`PASSWORD NULL` | Let the role [authenticate their access to a secure cluster](authentication.html#client-authentication) using this new password. Passwords should be entered as a [string literal](sql-constants.html#string-literals). For compatibility with PostgreSQL, a password can also be entered as an identifier. <br><br>To prevent a role from using [password authentication](authentication.html#client-authentication) and to mandate [certificate-based client authentication](authentication.html#client-authentication), [set the password as `NULL`](#prevent-a-role-from-using-password-authentication).
-`VALID UNTIL timestamp`/`VALID UNTIL NULL` | The date and time (in the [`timestamp`](timestamp.html) format) after which the password is not valid.
+`SQLLOGIN`/`NOSQLLOGIN` | Allow or disallow a role to log in using the SQL CLI with one of the [client authentication methods](authentication.html#client-authentication). Setting the role option to `NOSQLLOGIN` prevents the role from logging in using the SQL CLI with any authentication method while retaining the ability to log in to DB Console. It is possible to have both `NOSQLLOGIN` and `LOGIN` set for a role and `NOSQLLOGIN` takes precedence on restrictions.
+`VALID UNTIL` |  The date and time (in the [`timestamp`](timestamp.html) format) after which the [password](#parameters) is not valid.
+`VIEWACTIVITY`/`NOVIEWACTIVITY` | Allow or disallow a role to see other roles' [queries](show-statements.html) and [sessions](show-sessions.html) using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the [Statements](ui-statements-page.html) and [Transactions](ui-transactions-page.html) pages in the DB Console. With `NOVIEWACTIVITY`, the `SHOW` commands show only the role's own data and the DB Console pages are unavailable. If a user has the `VIEWACTIVITYREDACTED` role, they will be able to see DB Console pages. <br><br>By default, the role option is set to `NOVIEWACTIVITY` for all non-admin roles.
+`VIEWACTIVITYREDACTED`/`NOVIEWACTIVITYREDACTED` | Allow or disallow a role to see other roles' queries and sessions using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the Statements and Transactions pages and the [statements diagnostics bundle](ui-statements-page.html#diagnostics), which can contain PII information, in the DB Console. With `NOVIEWACTIVITYREDACTED`, the user won't have access to the usage of statements diagnostics bundle in the DB Console. It is possible to have both `NOVIEWACTIVITY` and `NOVIEWACTIVITYREDACTED`, and `NOVIEWACTIVITYREDACTED` takes precedence on restrictions. If the user has `VIEWACTIVITY` but doesn't have `VIEWACTIVITYREDACTED`, they will be able to see DB Console pages. <br><br>By default, the role option is set to `NOVIEWACTIVITYREDACTED` for all non-admin roles.
```

Comment:
You added the NO there assuming by mistake.
- With `VIEWACTIVITYREDACTED`, the user won't have access to the...
- It is possible to have both `VIEWACTIVITY` and `VIEWACTIVITYREDACTED`, and `VIEWACTIVITYREDACTED` takes precedence on restrictions

Reply:
It was deliberate. To me the roles should be named in a parallel structure. If you can't do something the role should start with NO. I find it super confusing how these roles work.

Reply:
`VIEWACTIVITYREDACTED`/`NOVIEWACTIVITYREDACTED` | Allow or disallow a role to see other roles' queries and sessions using `SHOW STATEMENTS`, `SHOW SESSIONS`, and the Statements and Transactions pages in the DB Console. With `VIEWACTIVITYREDACTED`, the user won't have access to the usage of statements diagnostics bundle, which can contain PII information, in the DB Console. It is possible to have both `VIEWACTIVITY` and `VIEWACTIVITYREDACTED`, and `VIEWACTIVITYREDACTED` takes precedence on restrictions. If the user has `VIEWACTIVITY` but doesn't have `VIEWACTIVITYREDACTED`, they will be able to see DB Console pages and have access to the usage of statements diagnostics bundle. <br><br>By default, the role option is set to `NOVIEWACTIVITYREDACTED` for all non-admin roles.